server-setup/tasks-handlers.ts

```diff
@@ -26,10 +60,36 @@ const handlers = {
     const dataType = task.id;
     const dataVersion = server.version;
 
+    const client: FhirClient = new FetchFhirClient({
+      baseUrl: server.url,
+    });
+
+    if (
+      await isTerminologySourceImported(client, {
+        source: dataSource,
+        system: dataType,
+        version: dataVersion,
+      })
+    ) {
+      console.log(
+        `ℹ️ Code system ${task.id} version ${server.version} from ${task.source} already ingested`
+      );
+      return;
+    }
+
     const { stdout, stderr, exitCode } =
       await $`hapi-fhir-cli upload-terminology -d "${TERMINOLOGIES_DATA_BASEPATH}${dataSource}" -v "${dataVersion}" -t "${server.url}" -u "${dataType}"`;
-
     logResults(stdout, stderr, exitCode);
+
+    createApplicationImportAuditEvent(
+      client,
+      makeAuditEventEntity({
+        source: dataSource,
+        system: dataType,
+        version: dataVersion,
+      }),
```

Comment:
can we abstract away the creation of the entity in `createApplicationImportAuditEvent`, the caller doesn't need to know about that, providing the significative entities properties should be sufficient

Reply:
✅